aries_cloudagent/connections/models/diddoc_v2/diddoc.py

```diff
@@ -239,11 +268,99 @@ def id(self, value: str) -> None:
-        verification_type="publicKey",
+    def add_verification_method(
+            self,
+            type: Union[PublicKeyType, str],
+            controller: Union[str, Sequence],
+            value: str,
+            *,
+            ident: str = None,
+            usage: str = None,
+            authentication: bool = False,
+            verification_type: str = "publicKey",
+            upsert: bool = False
+    ) -> VerificationMethod:
+        """Add a verification method to this document."""
+        if ident:
+            id = "{}#{}".format(self.id, ident)
+            if self._index.get(id) and not upsert:
+                raise ValueError("ID already exists, use arg upsert to update it")
+        else:
+            for index in range(1, 100):
+                id_aux = "{}#keys-{}".format(self.id, index)
+                if not self._index.get(id_aux):
+                    id = id_aux
+                    break
+
+        key = VerificationMethod(id, type, controller, usage, value, authentication)
+        self._set(key, upsert=upsert, verification_type=verification_type)
+        return key
+
+    def add_service(
+            self,
+            type: Union[str, list],
+            endpoint: Union[str, Sequence, dict],
+            ident: str = None,
+            priority: int = 0,
+            upsert: bool = False
+    ) -> Service:
+        """Add service to this document."""
+        if ident:
+            id = "{}#{}".format(self.id, ident)
+            if self._index.get(id) and not upsert:
+                raise ValueError("ID already exists, use arg upsert to update it")
+        else:
+            for index in range(1, 100):
+                id_aux = "{}#service-{}".format(self.id, index)
+                if not self._index.get(id_aux):
+                    id = id_aux
+                    break
+
+        service = Service(id, type, endpoint, priority=priority)
+        self._set(service, upsert=upsert)
+
+    def _add_keys(self, keys):
+        result = []
+        if isinstance(keys, list):
+            for key in keys:
+                result.append(key.id)
+                self._set(key)
+
+        elif isinstance(keys, VerificationMethod):
+            result.append(keys.id)
+            self._set(keys)
+
+        return result
+
+    def add_didcomm_service(
+            self,
+            type: Union[str, list],
+            recipient_keys: Union[Sequence[VerificationMethod], VerificationMethod],
+            routing_keys: Union[Sequence[VerificationMethod], VerificationMethod],
+            endpoint: Union[str, Sequence, dict],
+            priority: int = 0
+    ) -> Service:
+        """Add DIDComm Service to this document."""
+        for index in range(1, 100):
+            id_aux = "{}#didcomm-{}".format(self.id, index)
+            if not self._index.get(id_aux):
+                id = id_aux
+                break
+
+        # RECIPIENT KEYS
+        recip_keys = self._add_keys(recipient_keys)
```

Comment:
I think the usage of this API should require that verification methods used in the DIDComm service must first be added to the document through the `add_verification_method` method. It should therefore be an error condition if the passed verification method is not already found on the document.